### DISCUSSION TITLE
Should SwaggerForOcelot support DownstreamPathTemplate querystring params?

### DISCUSSION THREAD

**Discussion by clockworkmice** [Burgyn/MMLib.SwaggerForOcelot#236] [Q&A]
Could you tell me if SwaggerForOcelot is supposed to support downstream querystring mapping? I can see it mentioned in unit tests but can't get it to work myself.

My scenarios is I have an upstream that includes a path param `/example/dogs/{id}/type` and a downstream that expects that param as a querystring `/types?id={id}`.

If I take the querystring out of the `DownstreamPathTemplate` the swagger mapping works (but the param isn't passed downstream so it doesn't function correctly). If I put the querystring in the `DownstreamPathTemplate` the call functions but the swagger no longer maps.

I have downstream route like this:
`app.MapGet("/types", ([FromQuery] int id) => new { id, Name = $"Type#{id}" });`

And Ocelot setup with a `DownstreamPathTemplate` set up to take a path param and put it in a querystring param like this (functions but ignored in swagger docs):
```{
  "SwaggerKey": "example",

  "UpstreamPathTemplate": "/example/dogs/{id}/type",
  "UpstreamHttpMethod": [ "Get" ],

  "DownstreamPathTemplate": "/types?id={id}",
  "DownstreamScheme": "http",
  "DownstreamHostAndPorts": [
    {
      "Host": "example.myapi.com",
      "Port": 80
    }
  ]
},
```

If I switch the above line to `"DownstreamPathTemplate": "/types",` the swagger docs work but id isn't passed downstream and so the feature doesn't work.

The downstream swagger.json contains this section:
```"/types": {
      "get": {
        "tags": [
          "ExampleDownstream.Api"
        ],
        "parameters": [
          {
            "name": "id",
            "in": "query",
            "required": true,
            "schema": {
              "type": "integer",
              "format": "int32"
            }
          }
        ],
        "responses": {
          "200": {
            "description": "Success",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/Int32String<>f__AnonymousType0"
                }
              }
            }
          }
        }
      }
    }```

**Comment by Burgyn** [comment]
Hi,

unfortunately, currently the translating parameter is not supported.  See issue #197.
